src/lib/viewers/controls/page/PageControls.tsx

```diff
@@ -0,0 +1,129 @@
+    const [isInputShown, setIsInputShown] = React.useState(false);
+    const [tempPageNumber, setTempPageNumber] = React.useState(pageNumber.toString());
+    const inputEl = React.useRef<HTMLInputElement>(null);
+
+    React.useEffect(() => {
+        if (inputEl && inputEl.current) {
+            setTempPageNumber(pageNumber.toString());
+            inputEl.current.focus();
+        }
+    }, [isInputShown, pageNumber]);
+
+    const onPageNumInputChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
+        setTempPageNumber(e.target.value);
+    };
+
+    const onPageNumInputBlur = (): void => {
+        const newPageNumber = parseInt(tempPageNumber, 10);
+
+        if (!Number.isNaN(newPageNumber)) {
+            onPageChange(newPageNumber);
+        }
+
+        setIsInputShown(false);
+    };
+
+    const onPageNumInputKeydown = (e: React.KeyboardEvent<HTMLInputElement>): void => {
+        const key = decodeKeydown(e);
+
+        switch (key) {
+            case 'Enter':
+            case 'Tab':
+                viewer.focus();
+                // The keycode of the 'next' key on Android Chrome is 9, which maps to 'Tab'.
+                // We normally trigger the blur handler by blurring the input
+                // field, but this doesn't work for IE in fullscreen. For IE,
+                // we blur the page behind the controls - this unfortunately
+                // is an IE-only solution that doesn't work with other browsers
+
+                if (Browser.getName() !== BROWSERS.INTERNET_EXPLORER && e.target) {
+                    (e.target as HTMLInputElement).blur();
+                }
+
+                e.stopPropagation();
+                e.preventDefault();
+                break;
+
+            case 'Escape':
+                setIsInputShown(false);
+                viewer.focus();
+
+                e.stopPropagation();
+                e.preventDefault();
+                break;
+            default:
+                break;
+        }
+    };
+
+    return (
+        <div className="bp-PageControls">
+            <button
+                className="bp-PageControls-button"
+                data-testid="bp-PageControls-previous"
+                disabled={pageNumber === 1}
+                onClick={(): void => onPageChange(pageNumber - 1)}
+                title={__('previous_page')}
+                type="button"
+            >
+                <IconArrowUp24 />
+            </button>
+            {!isInputShown && (
+                <button
+                    aria-label={__('enter_page_num')}
+                    className="bp-PageControls-num-button"
+                    onClick={(): void => setIsInputShown(true)}
+                    title={__('enter_page_num')}
+                    type="button"
+                >
+                    <span className="bp-PageControls-num-button-label">{`${pageNumber} / ${pageCount}`}</span>
```

Comment:
Is this inner label necessary or can we center the text within the button itself?

Reply:
Yes, there's an area around the text that's colored differently